cohort-engine/src/test/java/com/ibm/cohort/engine/BaseFhirTest.java

```diff
@@ -170,18 +185,53 @@ protected CapabilityStatement getCapabilityStatement() {
 		metadata.setFhirVersion(Enumerations.FHIRVersion._4_0_1);
 		return metadata;
 	}
+	
+
+	protected CqlEngineWrapper setupTestFor(Patient patient, String... elm) throws Exception {
+		IBMFhirServerConfig fhirConfig = new IBMFhirServerConfig();
+		fhirConfig.setEndpoint("http://localhost:" + HTTP_PORT);
+		fhirConfig.setUser("fhiruser");
+		fhirConfig.setPassword("change-password");
+		fhirConfig.setTenantId("default");
+
+		return setupTestFor(patient, fhirConfig, elm);
+	}
 
-	protected CqlEngineWrapper setupTestFor(Patient patient, FhirServerConfig fhirConfig, String... elm)
+	protected CqlEngineWrapper setupTestFor(Patient patient, FhirServerConfig fhirConfig, String... resources)
 			throws Exception {
 
 		mockFhirResourceRetrieval("/metadata", getCapabilityStatement());
 		mockFhirResourceRetrieval(patient);
 
+		// This is a hack to get a single CQL resource from the provided set
+		// of library resources. This will be translated as needed.
+		final AtomicReference<String> cql = new AtomicReference<>();
```

Comment:
Pedantic: Is there any reason to use `AtomicReference` here?  Are we doing some form of multithreading in our tests?

Reply:
AtomicReference is used as a container for a String so that it can be final, but the contained object can change. This enables use inside nested classes.